BinaryEncoding.md

```diff
@@ -234,6 +233,45 @@ This section must be preceded by a [Functions](#functions-section) section.
 | count | `varuint32` | count of entries to follow |
 | entries | `uint16*` | repeated indexes into the function table |
 
+### Names section
+
+ID: `names`
+
+This section may occur 0 or 1 times and does not change execution semantics. A
+validation error in this section does not cause validation for the whole module
+to fail and is instead treated as if the section was absent. The expectation is
+that, when a binary WebAssembly module is viewed in a browser or other
+development environment, the names in this section will be used as the names of
+functions and locals in the [text format](TextFormat.md).
+
+| Field | Type | Description |
+| ----- |  ----- | ----- |
+| count | `varuint32` | count of names to follow |
+| names | `function_names*` | sequence of names |
+
+The sequence of `function_name` assigns names to the corresponding
+function index. The count may be greater or less than the actual number of
```

Comment:
Nothing specific, just thinking: if we're going to drop the equality requirement, might as well drop it all the way.